src/metabase/api/card.clj

```diff
@@ -125,6 +128,20 @@
   [_filter-option model-id]
   (cards-for-segment-or-metric :segment model-id))
 
+(defmethod cards-for-filter-option* :stale
+  [_]
+  (let [card-id->field-usages (group-by :card_id (t2/select :model/FieldUsage
+                                                            {:select [:fu.* [:f.name :new_column_name] [:t.name :new_table_name]]
+                                                             :from   [[(t2/table-name :model/FieldUsage) :fu]]
+                                                             :join   [[(t2/table-name :model/Field) :f] [:= :f.id :fu.field_id]
+                                                                      [(t2/table-name :model/Table) :t] [:= :t.id :f.table_id]]
+                                                             :where  [:= false :fu.is_current]}))
+        card-ids              (keys card-id->field-usages)]
+    (if (empty? card-ids)
+      []
+      (map #(assoc % :field_usages (get card-id->field-usages (:id %)))
```

Comment:
nitpick: I'd just lean on function application of the map, as its name encourages us to think of it as an opaque function.